src/port.js

```diff
@@ -114,6 +141,23 @@ export default class Port {
 		}
 
 	}
+	registerService(serviceType, version, service) {
+		if(this.isConnected) {
+			throw new Error('Register services before connecting');
+		}
+		if(!typeNameValidator.test(serviceType)) {
+			throw new Error(`Invalid service type "${serviceType}"`);
+		}
+		let methodNames = [];
+		for(let p in service) {
+			if(typeof(service[p]) === 'function') {
+				methodNames.push(p);
+				this.onRequest(`service:${serviceType}:${version}:${p}`, service[p]);
```

Comment:
Is it worth noting in docs somewhere that calling `registerService` and then mutating `service` will lead to undefined behaviour?  E.g.:

```
let service = { 
    add: (a, b) => a + b
};
client.registerService('calc', '1.0', service);
service.add = null;
// Any calls to `calc.add` will explode.
```